localstack/services/awslambda/invocation/version_manager.py

```diff
@@ -404,7 +406,15 @@ def invocation_loop(self, *args, **kwargs) -> None:
 
                 environment = None
                 # TODO avoid infinite environment spawning retrying
+                environment_start_attempt_nr = 0
                 while not environment:
+                    environment_start_attempt_nr += 1
```

Comment:
That counter increases without actually creating a new environment. I only got one container despite `LAMBDA_REMOVE_CONTAINERS=0` and got the exception "Exceeded maximum retries ..."